### PR TITLE
[ARCTIC-1380] [core] fix NPE when scan entries of Iceberg table < 1.0.0

### LINKED ISSUE #1380: [Bug]: Occur NPE when Iceberg version upgrade to 1.1.0
### What happened?

Iceberg version upgrade to 1.1.0, I meet the error.

relevant PR #989 

### Affects Versions

master

### What engines are you seeing the problem on?

AMS

### How to reproduce

_No response_

### Relevant log output

```shell
2023-04-19 18:39:16,966 ERROR [Metastore Scheduled Check Worker 5] [com.netease.arctic.ams.server.optimize.OptimizeCheckTask] [] - unexpected check error catalog.database.table
java.lang.NullPointerException: null


2023-04-20 13:47:21,977 ERROR [Metastore Scheduled Orphan Worker 1] [com.netease.arctic.ams.server.service.impl.OrphanFilesCleanService] [] - catalog.database.table orphan file clean unexpected error
java.lang.NullPointerException: null
        at com.netease.arctic.scan.TableEntriesScan.lambda$entries$0(TableEntriesScan.java:194) ~[arctic-core-0.5.0-SNAPSHOT.jar:?]
        at org.apache.iceberg.io.CloseableIterable$7$1.next(CloseableIterable.java:202) ~[iceberg-api-1.1.0.jar:?]
        at org.apache.iceberg.io.FilterIterator.advance(FilterIterator.java:65) ~[iceberg-api-1.1.0.jar:?]
        at org.apache.iceberg.io.FilterIterator.hasNext(FilterIterator.java:49) ~[iceberg-api-1.1.0.jar:?]
        at com.netease.arctic.ams.server.utils.UnKeyedTableUtil.getAllContentFilePath(UnKeyedTableUtil.java:58) ~[arctic-ams-server-0.5.0-SNAPSHOT.jar:?]
        at com.netease.arctic.ams.server.service.impl.OrphanFilesCleanService.getValidContentFiles(OrphanFilesCleanService.java:321) ~[arctic-ams-server-0.5.0-SNAPSHOT.jar:?]
        at com.netease.arctic.ams.server.service.impl.OrphanFilesCleanService.cleanContentFiles(OrphanFilesCleanService.java:128) ~[arctic-ams-server-0.5.0-SNAPSHOT.jar:?]
        at com.netease.arctic.ams.server.service.impl.OrphanFilesCleanService$TableOrphanFileClean.run(OrphanFilesCleanService.java:114) ~[arctic-ams-server-0.5.0-SNAPSHOT.jar:?]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) ~[?:1.8.0_141]
        at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308) ~[?:1.8.0_141]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180) ~[?:1.8.0_141]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294) ~[?:1.8.0_141]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) ~[?:1.8.0_141]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) ~[?:1.8.0_141]
        at java.lang.Thread.run(Thread.java:748) ~[?:1.8.0_141]
```

### Anything else

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by wangtaohz**
## Why are the changes needed?

fix #1380 

## Brief change log

  - support sequence number from entry to be null

## How was this patch tested?
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

## Documentation

  - Does this pull request introduces a new feature? (**yes** / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / **not documented**)

### CODE REVIEW

File: core/src/main/java/com/netease/arctic/IcebergFileEntry.java [hunk 1]
```diff
@@ -18,45 +18,51 @@
...
 
-  public void setFile(ContentFile<?> file) {
-    this.file = file;
+  public ManifestEntryFields.Status getStatus() {
+    return status;
+  }
+
+  @Override
+  public String toString() {
+    return "IcebergFileEntry{" +
```
Comment (zhoujinsong) [inline core/src/main/java/com/netease/arctic/IcebergFileEntry.java:61]: How about using `Objects.toStringHelpers` template to generate `toString` method as we are dependent on guava lib. DEA has already implemented this template, so you just need to select it when generating the toString method.

Comment (wangtaohz) [reply core/src/main/java/com/netease/arctic/IcebergFileEntry.java]: OK, I refactor it.

**Review by baiyangtx — APPROVED** [review]
LGTM

**Comment by baiyangtx** [comment]
I have been reviewed the code and all is fine to me. 

@zhoujinsong will you need to review it? I keep the PR un-merged.

**Review by zhoujinsong — APPROVED** [review]
LGTM.

**Pull request merged**